src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -336,12 +337,19 @@ private void WatchForFileSystemEventsThreadStart(object arg)
 
                 for (long i = 0; i < numEvents.ToInt32(); i++)
                 {
-                    Debug.Assert(eventPaths[i].Length > 0, "Empty events are not supported");
-                    Debug.Assert(eventPaths[i][eventPaths[i].Length - 1] != '/', "Trailing slashes on events is not supported");
+                    Debug.Assert(eventPaths[i] != null);
+                    int endPoint = new ReadOnlySpan<byte>(eventPaths[i], 255).IndexOf<byte>(0);
```

Comment:
Why are you limiting this to 255? Aren't these full paths?

Reply:
These are full paths, we will have to supply some boundary length for them. should we use some other constant ?

Reply:
Why do you have to supply a boundary length? Paths can be up to 4K in known cases, but theoretically can be more. Don't try and wrap a span around it when you don't know where the buffer ends- increment the pointer.